content/algorithms/flow/dinitz_alg.md

```diff
@@ -0,0 +1,687 @@
+    flow_value += this_flow
+
+R.graph["flow_value"] = flow_value
+plot_G('current')
+```
+
+```
+def gif(filename, array, fps, scale=1.0):
+    fname, _ = os.path.splitext(filename)
+    filename = fname + '.gif'
+    clip = ImageSequenceClip(array, fps=fps).resize(scale)
+    clip.write_gif(filename, fps=fps)
+    return clip
+```
+
+```
+limage = []
+for i in range(1,36):
+    limage.append(asarray(Image.open(os.path.join("images","gif",f"final_{i}.jpg"))))
+```
+
+```
+gif("images/example",limage,fps=0.5)
+```
+````
+
+```{code-cell} ipython3
+%matplotlib inline
+import networkx as nx
+import matplotlib.pyplot as plt
+import pickle
+```
+
+```{code-cell} ipython3
+gname = "egnetwork"
+# loading the graph
+G = nx.read_graphml(f"data/{gname}.graphml")
+with open(f"data/pos_{gname}", "rb") as fp:
+    pos = pickle.load(fp)
+```
+
+```{code-cell} ipython3
+fig, axes = plt.subplots(4, 2, figsize=(20, 30))
+
+# assign colors and labels to nodes based on their type
+color_map = {"t": "skyblue", "s": "skyblue"}
+node_colors = [color_map[u] if u in color_map.keys() else "0.8" for u in G.nodes]
+node_labels = {u: u for u in G.nodes}
+cutoff_list = [5, 10, 15, 20, 25, 30, 35, 40]
+
+for i in range(8):
+
+    # calculating the maximum flow with the cutoff value
+    R = nx.flow.dinitz(G, s="s", t="t", capacity="capacity", cutoff=cutoff_list[i])
+
+    # coloring and labeling edges depending on if they have non-zero flow value or not
+    edge_colors = ["0.8" if R[u][v]["flow"] == 0 else "0" for u, v in G.edges]
+    edge_labels = {
+        (u, v): f"{R[u][v]['flow']}/{G[u][v]['capacity']}"
+        for u, v in G.edges
+        if R[u][v]["flow"] != 0
+    }
+
+    # drawing the network
+    nx.draw_networkx_nodes(
+        G, pos=pos, ax=axes[i // 2][i % 2], node_size=500, node_color=node_colors
+    )
+    nx.draw_networkx_labels(
+        G, pos=pos, ax=axes[i // 2][i % 2], labels=node_labels, font_size=14
+    )
+    nx.draw_networkx_edges(G, pos=pos, ax=axes[i // 2][i % 2], edge_color=edge_colors)
+    nx.draw_networkx_edge_labels(
+        G, pos=pos, ax=axes[i // 2][i % 2], edge_labels=edge_labels, font_size=14
+    )
+    axes[i // 2][i % 2].set_title(
+        f"Max Flow = {R.graph['flow_value']}\nCutoff value of = {cutoff_list[i]}",
+        size=15,
+    )
+
+fig.tight_layout()
+```
+
+Note: Iteration are stopped if the maximum flow found so far exceeds the cutoff value
+## Reductions and Applications
+There are many other problems which can be reduced to Maximum flow problem for eg.
+* [Maximum Bipartite Matching](https://en.wikipedia.org/wiki/Matching_(graph_theory))
+* [Assignment Problem](https://en.wikipedia.org/wiki/Assignment_problem)
+* [Transportation Problem](https://en.wikipedia.org/wiki/Transportation_theory_(mathematics))
+
+and many others
+
+Note that even though dinitz works in $O(n^2m)$ strongly polynomial time, i.e. to say it 
+doesn't depend on the value of flow. It is noteworthy that its performance of biparted 
```

Comment:
```suggestion
Note that dinitz works in $O(n^2m)$ strongly polynomial time, that is to say it 
doesn't depend on the value of flow. It is noteworthy that its performance of biparted 
```
The wording here was a bit off... I *think* the suggestion captures what you were trying to say, but I could be wrong. Please correct me as you see fit!